oidc-authenticator/src/main/java/org/xwiki/contrib/oidc/auth/OIDCAuthServiceImpl.java

```diff
@@ -103,6 +172,33 @@ public XWikiUser checkAuth(XWikiContext context) throws XWikiException
         return user;
     }
 
+    private XWikiUser checkAccessToken(String authorizationHeaderValue) {
+        JWTClaimsSet jwtClaimsSet;
+        try {
+            jwtClaimsSet = jwtProcessor.process(authorizationHeaderValue.substring("Bearer ".length()), null);
+
+            ClaimsSet claimsSet = new ClaimsSet();
+            claimsSet.putAll(jwtClaimsSet.toJSONObject());
+            
+            UserInfo userInfo = new UserInfo(jwtClaimsSet);
```

Comment:
I don't really understand your process here. You don't seem to ever validate the token on the provider and seems to expect it to be a complete (signed) user info, which is not really what an access token is supposed to be in OIDC AFAIK.

Reply:
I looked at the code again. To get full compatibility with the existing login logic, I propose the following approach:
- Send both an id and access token using `X-Id-Token` and `X-Access-Token` headers
- Validate the signature and audience of the Id token (like I did in this PR for the access token)
- Use the access token to retrieve the user info
What do you think?

Reply:
I implemented the mechanism above. Please review.